### PR TITLE
BaSP-MR-121: Members redux put post

### PR DESCRIPTION
**Pull request by EmaBaldor**
Added redux logic to put and post member.
Completed thunks, actions, constans and reducers files.
The logic of the form was modified to use the data from the state.

### CODE REVIEW

**Comment by EmaBaldor** [comment]
[[Members] - Move post and put logic.](https://trello.com/c/piBP9uwG/121-members-move-post-and-put-logic)

File: src/Components/Members/Form/index.jsx [hunk 1]
```diff
@@ -26,84 +30,64 @@ const MemberForm = () => {
...
-    setMember({
-      ...member,
-      [e.target.name]: e.target.value
-    });
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (id) {
+      const memberUpdate = { ...member };
+      delete memberUpdate._id;
+      delete memberUpdate.__v;
```
Comment (fernandofernandezuy) [inline src/Components/Members/Form/index.jsx]: You can do this when you set the member in the memberById() function, avoiding to declare a new constant

File: src/Components/Members/Form/index.jsx [hunk 2]
```diff
@@ -26,84 +30,64 @@ const MemberForm = () => {
...
   };
 
-  const getMemberById = async (id) => {
+  const memberUpdateFunction = async (id, member) => {
     try {
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/members/${id}`);
-      const data = await response.json();
-      setMember(data.data);
+      const { data } = await dispatch(updateMember(id, member));
+      setAlertMessage(`Member ${data.firstName} ${data.lastName} updated.`);
```
Comment (fernandofernandezuy) [inline src/Components/Members/Form/index.jsx]: You can set the alert message using the response of the server, data.message

File: src/Components/Members/Form/index.jsx [hunk 3]
```diff
@@ -26,84 +30,64 @@ const MemberForm = () => {
...
-        setAlertMessage(data.message);
-        setIsSuccess(false);
-        setShowAlert(true);
-      } else {
-        setAlertMessage(data.message);
-        setIsSuccess(true);
-        setShowAlert(true);
-      }
+      const { data } = await dispatch(addMember(member));
+      setAlertMessage(`Member ${data.firstName} ${data.lastName} created.`);
```
Comment (fernandofernandezuy) [inline src/Components/Members/Form/index.jsx]: Same here

**Review by Federico-Peralta — APPROVED** [review]
Tested!

File: src/Components/Members/index.jsx [hunk 4]
```diff
@@ -10,39 +10,44 @@ import { useSelector, useDispatch } from 'react-redux';
...
 
 const Members = () => {
+  const dispatch = useDispatch();
   const history = useHistory();
   const members = useSelector((state) => state.members.data.data);
   const loading = useSelector((state) => state.members.loading);
-  const [showWarning, setShowWarning] = useState(false);
   const [showAlert, setShowAlert] = useState(false);
-  const [isSuccess, setIsSuccess] = useState(false);
+  const [typeStyle, SetTypeStyle] = useState('');
```
Comment (AleMac17) [inline src/Components/Members/index.jsx]: We should use camelCase to nominate our variables. In this case -setTypeStyle-

File: src/Components/Members/index.jsx [hunk 5]
```diff
@@ -10,39 +10,44 @@ import { useSelector, useDispatch } from 'react-redux';
...
+    setIsDelete(false);
+    setShowAlert(false);
     try {
-      const data = await dispatch(deleteMember(memberToDelete));
-      setAlertMessage(data.message);
-      setIsSuccess(true);
+      const { data } = await dispatch(deleteMember(memberToDelete));
+      setAlertMessage(`Member ${data.firstName} ${data.lastName} deleted.`);
+      setTitle('Success');
+      SetTypeStyle('success');
```
Comment (AleMac17) [inline src/Components/Members/index.jsx]: We should use camelCase to nominate our variables. In this case -setTypeStyle-

File: src/Components/Members/index.jsx [hunk 6]
```diff
@@ -10,39 +10,44 @@ import { useSelector, useDispatch } from 'react-redux';
...
       setShowAlert(true);
-      setShowTable(true);
     } catch (error) {
       setAlertMessage(error.message);
-      setIsSuccess(false);
+      setTitle('Error');
+      SetTypeStyle('error');
```
Comment (AleMac17) [inline src/Components/Members/index.jsx]: We should use camelCase to nominate our variables. In this case -setTypeStyle-

**Review by AleMac17 — CHANGES_REQUESTED** [review]

**Review by AleMac17 — APPROVED** [review]

**Comment by araefedg** [comment]
The functionality works correctly, Great Job!

**Pull request merged**